src/collection.ts

```diff
@@ -684,10 +681,12 @@ export class Collection<TSchema extends Document = Document> {
     indexes: string | string[],
     options?: IndexInformationOptions
   ): Promise<boolean> {
-    return executeOperation(
-      this.client,
-      new IndexExistsOperation(this as TODO_NODE_3286, indexes, resolveOptions(this, options))
-    );
+    const indexNames: string[] = [indexes].flat();
+    const allIndexes: string[] = await this.listIndexes(options)
+      .map(({ name }) => name)
+      .toArray();
+
+    return indexNames.every(name => allIndexes.includes(name));
```

Comment:
Fine with all of the above, leaving it as-is seems least preferable since we're listing _all_ indexes I don't want to assume how many that could be, so seems worth paying an upfront cost of hashing the values to make `every` a bit snappier but I don't feel too strongly

Does this change make this PR a fix?

Reply:
> Fine with all of the above, leaving it as-is seems least preferable since we're listing all indexes I don't want to assume how many that could be, so seems worth paying an upfront cost of hashing the values to make every a bit snappier but I don't feel too strongly

I modified the implementation to use a set.  But I do think it's worth keeping in mind that index management helpers are not typically used in hot code paths, they're usually administrative helpers that are used in tooling / scripting.